node/node.go

```diff
@@ -220,7 +221,13 @@ func (n *Node) Run(ctx context.Context) {
 	synchronizer := sync.New(n.blockchain, adaptfeeder.New(client), n.log, n.cfg.PendingPollInterval)
 	gatewayClient := gateway.NewClient(n.cfg.Network.GatewayURL(), n.log)
 
-	http := makeHTTP(n.cfg.RPCPort, rpc.New(n.blockchain, synchronizer, n.cfg.Network, gatewayClient, n.version, n.log), n.log)
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", n.cfg.RPCPort))
+	if err != nil {
+		n.log.Errorw("Failed to listen for RPC requests", "port", n.cfg.RPCPort)
+		return
+	}
+	defer listener.Close()
+	http := makeHTTP(listener, rpc.New(n.blockchain, synchronizer, n.cfg.Network, gatewayClient, n.version, n.log), n.log)
```

Comment:
I am not sure about this, our `Run` function is already overly crowded as it is.

Instead of changing `rpc.New`, can we introdue something like `rpc.NewWithListener` that we can use in tests only?

Reply:
I learned that don't need to close the listener (`http.Shutdown` does it for us), which decreases complexity and size of the function.